src/framework/auth/strategies/oauth2/oauth2-strategy.ts

```diff
@@ -376,6 +377,16 @@ export class NbOAuth2AuthStrategy extends NbAuthStrategy {
     }, {}) : {};
   }
 
+  protected createRefreshedToken(res, existingToken: NbAuthRefreshableToken): NbAuthToken {
```

Comment:
I believe we can leave this check as:

```
if (!refreshedToken.getRefreshToken() && existingToken.getRefreshToken()) {
```
as we can only have the `NbAuthRefreshableToken` token here, which presumes that both required get and set method exist because this method could only be called from the `refreshToken` one, which cannot be called for a non-refreshable token.
In case when the developer forces a call of `refreshToken` method for a non-refreshble token it will be a typescript level error, which is completely fine in this case.

Reply:
`!refreshedToken.getRefreshToken()` does not compile as `NbAuthToken` (the return type) does not implement this method.
That is the reason why i added the `isNbAuthRefreshableToken(refreshedToken)` test

So what do you prefer : 
- should we return NbAuthOauth2Token  ?
- or just cast in the body of the method the refreshedToken to NbAuth0Auth2Token ?
- or leave the code like in #10d0b31 commit ?

Reply:
@alain-charles sorry for the delay, wrote the comment last Thursday but forgot to submit.

Reply:
@nnixaa with the intersection type you propose above, the problem remains on `refreshedToken`

```const refreshedToken = this.createToken(res);```

`this.createToken(res)` returns  `NbAuthToken` that does not implement `getRefreshToken()`.

Any other idea ? 

Alain

Reply:
@alain-charles something like this probably? https://github.com/alain-charles/nebular/pull/1/files